test/runtests.jl

```diff
@@ -366,6 +366,17 @@ end
     threadlocal
   end
   @test eltype(settingabstype)<:AbstractFloat
+  # check for excessive allocations
+  function f(n)
+      @batch minbatch=10 threadlocal=1.0::Float64 for i in 1:n
+          threadlocal += 1.0/threadlocal
+      end
+      return threadlocal
+  end
+  f(1000)
+  b = @benchmark $f(1000) samples=1 evals=1
```

Comment:
You could test with `@allocated` to avoid the dependency on `BenchmarkTools`.
Something like
```julia
allocated(f::F) where {F} = @allocated f()
@test allocated(() -> f(1000)) < 300 + 40*Threads.nthreads()
```

Reply:
version bumped, `BenchmarkTools` removed, minor typo in readme fixed